### PR TITLE
[Merged by Bors] - feat: Lemma for `fderiv` of scalar function

### PR DESCRIPTION
**Pull request by adomasbaliuka**
Adds lemma for simplifying the Fréchet derivative when applied to a function maping scalars to scalars. 

This (currently named `fderiv_deriv'`) is a special case of `fderiv_deriv`. Neither of `simp`, `fun_prop`, `aesop`, `exact?` , `apply?` currently seems to make progress on it.

Possibly, it should be instead stated as `(fderiv 𝕜 f x : 𝕜 → 𝕜) = ((deriv f x) * ·)` if that makes it more likely to be applied by `simp`?

I think it should be marked `@[simp]` because one would basically always prefer the right hand side to the left hand side of the equality. Does that make sense?

I also think `fderiv_deriv` should be marked `@[simp]`, though maybe there is a reason why it isn't. Marking it such requires a change to the proof of `deriv_fderiv`, which was using a `simp` rather than a `simp only` (not a [non-terminal simp](https://leanprover-community.github.io/glossary.html#non-terminal-simp) but still to be avoided, I would assume?). Maybe other things are also likely to break, however, if those are added to simp?

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
Thanks for your contribution to mathlib! I guess you'd like code to be reviewed - hence I have labelled it `awaiting-review`. (You can read more about this and related labels [here](https://leanprover-community.github.io/contribute/index.html#lifecycle-of-a-pr).

File: Mathlib/Analysis/Calculus/Deriv/Basic.lean [hunk 1]
```diff
@@ -481,6 +481,12 @@ theorem fderiv_deriv : (fderiv 𝕜 f x : 𝕜 → F) 1 = deriv f x :=
 theorem deriv_fderiv : smulRight (1 : 𝕜 →L[𝕜] 𝕜) (deriv f x) = fderiv 𝕜 f x := by simp [deriv]
 #align deriv_fderiv deriv_fderiv
 
+@[simp]
+lemma fderiv_deriv' {f : 𝕜 → 𝕜} {x y : 𝕜} : (fderiv 𝕜 f x : 𝕜 → 𝕜) y = (deriv f x) * y := by
```
Comment (fpvandoorn) [inline Mathlib/Analysis/Calculus/Deriv/Basic.lean:485]:
```suggestion
lemma fderiv_eq_deriv_mul {f : 𝕜 → 𝕜} {x y : 𝕜} : (fderiv 𝕜 f x : 𝕜 → 𝕜) y = deriv f x * y := by
```

**Comment by fpvandoorn** [comment]
This lemma looks good to me. I would like a different name though (we want to avoid primes, and also the other two names are not great).

**Comment by fpvandoorn** [comment]
Actually, can you add a version for `f : 𝕜 → F` where the right-hand-side is `y \bu deriv f x`?

**Comment by fpvandoorn** [comment]
And it makes sense to me to make `fderiv_deriv` a simp-lemma. Feel free to do that. That might need some fixes in other places.

**Comment by adomasbaliuka** [comment]
I don't really understand why adding new `simp` lemmas breaks a lot proofs. How to judge if adding a `simp` lemma makes the situation overall better or worse?

**Comment by fpvandoorn** [comment]
Thanks!

bors merge

**Pull request closed without merging**